### ISSUE TITLE
Copy page doesn't trigger _create_page hook 

### ISSUE THREAD

**Issue by x-warrior** [wagtail/wagtail#3246]
### Issue Summary

Copying a Page doesn't trigger `*_create_page` hooks but essentially you're creating a page when copying

### Steps to Reproduce

1. Create a hook `after_create_page`
```
@hooks.register('after_create_page')
def test_hook_being_executed(request, page):
    print("test_hook_being_executed has been called")
``` 
2. Copy a page
3. The print message is not shown

When copying a page you're also creating a page, so maybe it would be nice to trigger _create_page hooks on copy action too? Or maybe we could create a new `after/before_copy_page` hook? 

What do you guys think?

**Comment by gasman** [comment]
Separate `*_copy_page` hooks would be more appropriate, I think. The existing `*_create_page` hooks are designed to intercept the 'create' action at the admin-view level - for example, they allow the developer to serve a custom response instead of redirecting back to the explorer. (The original use-case for `after_create_page` was to implement a 'limited' user type, who would be sent back to a custom dashboard view after creating a page.) At that level, 'create' and 'copy' are two very distinct actions.

(At the _model_ level, both actions involve page creation... but if you want to insert custom logic at that level, [signals](https://docs.djangoproject.com/en/1.10/ref/signals/#module-django.db.models.signals) are a better fit.)

**Comment by x-warrior** [comment]
@gasman May I add this *_copy_page hook based on _create_page hooks?

**Comment by gasman** [comment]
Yes please!

**Comment by gasman** [comment]
Implemented in #3271.
